File: Solution-Go/Longest_3.go
func lengthOfLongestSubstring(s string) int {
    freq := make(map[rune]int)
    l, repeat := 0, 0
    for _, c := range s {
        if freq[c] += 1; freq[c] == 2 {
            repeat += 1
        }
        if repeat > 0 {
            if freq[rune(s[l])] -= 1; freq[rune(s[l])] == 1 {
                repeat -= 1
            }
            l += 1
        }
    }
    return len(s) - l
}

File: Solution-Python/Longest_3.py
from collections import defaultdict

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        freq = defaultdict(int)
        l, repeat = 0, 0
        for r in range(len(s)):
            freq[s[r]] += 1
            if freq[s[r]] == 2:
                repeat += 1
            if repeat > 0:
                freq[s[l]] -= 1
                if freq[s[l]] == 1:
                    repeat -= 1
                l += 1
        return len(s) - l